lib/types.nix

```diff
@@ -275,6 +275,22 @@ rec {
       merge = mergeEqualOption;
     };
 
+    anyBool = mkOptionType {
+      name = "anyBool";
+      description = "boolean (any)";
+      descriptionClass = "noun";
+      check = isBool;
+      merge = loc: defs:
+        foldl'
+          (result: def:
+            # We don't need to worry about ||'s short-circuit behavior influencing strictness,
+            # because def.value is always evaluated before anyways
+            result || def.value
+          )
+          false
+          defs;
+    };
```

Comment:
Should `bool` have type merging instead?
`bias = true;` => or
`bias = false;` => and

Reply:
`bias = "quorum";` => chaos actually. Use `mkMerge` `replicate` to win.
:clown_face:

Reply:
I am a bit worried about backwards compat if we had something like `bool = boolWith { bias = null; }`, since the `name` of the type would change (unless we did `name = if bias == null then "bool" else "bool-with-${bias}"`). Though I guess the name shouldn't really influence anything, probably..

Other than that, I'm not sure if we'll ever even need the AND-based variant. Even the OR-based one here is a bit dubious. It's probably only really necessary because of the `functionTo bool` of `allowInsecurePredicate`, which we should actually be replaced with [RFC 127](https://github.com/NixOS/rfcs/blob/master/rfcs/0127-issues-warnings.md) instead anyways.

Actually considering that, it might make the most sense to declare this boolean type just locally in `pkgs/top-level/config.nix`. I don't know of any other use cases of it.

Reply:
I think combined process capability flags are a good candidate for the type. A service may have a set of security capabilities that should encompass all processes in that service.

Example:
- system default policy for any service: `cap.SYS_NET = mkDefault true;`, `cap.SYS_CHOWN = mkDefault false;`
- each programs sets its requests, both positive (need capability) as well as negative (do not need default capability):
  - program A: `cap.SYS_NET = false;`, `cap.SYS_CHOWN = false;`
  - program B: `cap.SYS_NET = false;`, `cap.SYS_CHOWN = true;`

`SYS_NET` shows the relevance of priorities, whereas `SYS_CHOWN` shows the need for `boolByOr`.

So I think it makes sense as a first class type.

Reply:
The `AND`-based variant would be for things that are enabled by default but can be disabled by any other module.
I suspect that these are mostly things that are phrased negatively.

`networking.firewall.enable` would be an example of a secretly negative option, although definitely not one that any module should just go and disable. It's negative, because enabling it prevents functionality.
I think it may be acceptable for a module to set `mkDefault false` on a nice to have feature that isn't as critical.
Maybe in general you want cpufreq enabled, but some hardware service that suffers from it may disable cpufreq, but only as a default.
So I guess AND is for all flags that are nice to enable by default but might have negative consequences, so that some other modules can disable it by default. All while letting the user pick a setting without `mkForce`.

Currently we may (ab)use the `mkOptionDefault`/`mkDefault` separation to get similar behavior.

Reply:
Alright let's have just `boolByOr` for now then, sounds fine to me